### PR TITLE
feat: Substrate Percentage Fee

### PR DESCRIPTION
**Pull request by FSM1**
## Description
Added a helper to fetch the fee type from the FeeRouter pallet.
Added logic to handle calculating the fee if the selected resource is configured to use the Percentage fee type. 

## Related Issue Or Context
Closes: #291 

## How Has This Been Tested? Testing details.
Fee is correctly calculated.
Tests were done on local env, with both min and max fee tested.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation

## Checklist:

- [X] I have commented my code, particularly in hard-to-understand areas.
- [X] I have ensured that all acceptance criteria (or expected behavior) from issue are met
- [ ] I have updated the documentation locally and in chainbridge-docs.
- [x] I have added tests to cover my changes.
- [X] I have ensured that all the checks are passing and green, I've signed the CLA bot

### CODE REVIEW

**Comment by FSM1** [comment]
@MakMuftic the code for handling % based fees is here, both for EVM and substrate.

No tests have been added yet.

I've tested the getFee method with percentage fee handler on substrate side, but not EVM.

Since I didn't have publish rights for the sygma-contracts npm package, I published a version under my own name space. Once this has been published correctly the import needs to be updated.

File: packages/sdk/src/chains/EVM/fee/percentageFee.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
+import { PercentageERC20FeeHandlerEVM__factory } from '@buildwithsygma/sygma-contracts';
+import { ethers } from 'ethers';
+import { formatBytes32String } from 'ethers/lib/utils';
```
Comment (wainola) [inline packages/sdk/src/chains/EVM/fee/percentageFee.ts:3]: this is to avoid having `utils.formatBytes32String`, isn't it?

File: packages/sdk/src/chains/EVM/fee/percentageFee.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+import { FeeHandlerType } from '../../../types/index.js';
+import { EvmFee } from '../types/index.js';
```
Comment (wainola) [inline packages/sdk/src/chains/EVM/fee/percentageFee.ts:5]: is it possible to have `import ... from '../types'`?

File: packages/sdk/src/chains/Substrate/__test__/deposit.test.ts [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import { ApiPromise } from '@polkadot/api';
 import { XcmMultiAssetIdType } from '../types';
 
-import * as Utils from '../utils/depositFns.js';
+import * as Utils from '../utils/depositFns';
```
Comment (wainola) [inline packages/sdk/src/chains/Substrate/__test__/deposit.test.ts:4]: why this change reference to the `.js` file, but it is present above?

File: packages/sdk/src/chains/Substrate/utils/getPercentageFee.ts [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  if (feeStructure.isNone) {
+    throw new Error('Error retrieving fee');
+  }
+
+  const [feeRate, min, max] = feeStructure
+    .unwrap()
+    .toArray()
+    .map(val => new BN(val.toString()));
+
+  // const fee = new BN(transfer.details.amount).
```
Comment (wainola) [inline packages/sdk/src/chains/Substrate/utils/getPercentageFee.ts:45]: remove commented code if we are not going to use this

**Review by wainola — CHANGES_REQUESTED** [review]

File: packages/sdk/src/chains/Substrate/__test__/deposit.test.ts (hunk 3 above)
Comment (FSM1) [reply packages/sdk/src/chains/Substrate/__test__/deposit.test.ts:4]: For some reason when using the automatic `update import` functionality, VSCode sometimes appends the `.js` suffix, but this is technically not required. Hence I am removing it.

**Review by wainola — APPROVED** [review]

File: examples/substrate-to-evm-fungible-transfer/package.json [hunk 5]
```diff
@@ -32,7 +32,7 @@
   },
   "dependencies": {
     "@buildwithsygma/sygma-sdk-core": "2.2.0",
-    "@polkadot/api": "10.7.3",
+    "@polkadot/api": "10.7.2",
```
Comment (mpetrun5) [inline examples/substrate-to-evm-fungible-transfer/package.json:35]: Why didn't we downgrade this?

**Review by mpetrun5 — APPROVED** [review]
LGTM

File: packages/sdk/src/chains/Substrate/utils/getPercentageFee.ts [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+ *   });
+ *
+ * @category Fee
+ * @param {ApiPromise} api - The Substrate API instance.
+ * @param {number} domainId - The ID of the domain.
+ * @param {XcmMultiAssetIdType} xcmMultiAssetId - The XCM MultiAsset ID of the asset. {@link https://github.com/sygmaprotocol/sygma-substrate-pallets#multiasset | More details}
+ * @returns {Promise<SubstrateFee>} A promise that resolves to a SubstrateFee object.
+ * @throws {Error} Unable to retrieve fee from pallet
+ */
+export const getPercentageFee = async (
```
Comment (mpetrun5) [inline packages/sdk/src/chains/Substrate/utils/getPercentageFee.ts:26]: Would be nice to have some tests for this

**Review by mpetrun5 — CHANGES_REQUESTED** [review]

File: packages/sdk/src/chains/Substrate/utils/getFeeHandler.ts [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+ * Retrieves the fee handler for a given domainId and asset.
+ *
+ * @category Fee
+ * @param {ApiPromise} api - The Substrate API instance.
+ * @param {number} domainId - The ID of the domain.
+ * @param {XcmMultiAssetIdType} xcmMultiAssetId - The XCM MultiAsset ID of the asset. {@link https://github.com/sygmaprotocol/sygma-substrate-pallets#multiasset | More details}
+ * @returns {Promise} A promise that resolves to a FeeHandlerType object.
+ * @throws {Error} Unable to retrieve fee from pallet
+ */
+export const getFeeHandler = async (
```
Comment (mpetrun5) [inline packages/sdk/src/chains/Substrate/utils/getFeeHandler.ts:15]: And this

**Review by mpetrun5 — APPROVED** [review]

**Review by MakMuftic — APPROVED** [review]

File: examples/substrate-to-evm-fungible-transfer/package.json (hunk 5 above)
Comment (FSM1) [reply examples/substrate-to-evm-fungible-transfer/package.json:35]: this was downgraded to match the version that was used in the rest of the SDK.

I tried updating to a newer version than 10.7.2 but there were some type incompatibilities that prevented the project from building.
